tests/view/writer/clear.js

```diff
@@ -0,0 +1,159 @@
+			);
+		} );
+
+		it( 'should remove matches element from range when range is inside text node', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>F{o<attribute:b>o</attribute:b> ba}r</container:p>',
+				'<container:p>Fo bar</container:p>'
+			);
+		} );
+
+		it( 'should remove matches elements from range', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>[Fo<attribute:b>o</attribute:b> b<attribute:b>a</attribute:b>r]</container:p>',
+				'<container:p>Fo br</container:p>'
+			);
+		} );
+
+		it( 'should remove matches elements from range when range is inside text node', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>F{o<attribute:b>o</attribute:b> <attribute:b>b</attribute:b>a}r</container:p>',
+				'<container:p>Fo ar</container:p>'
+			);
+		} );
+
+		it( 'should remove only matches element', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>F{o<attribute:i>o</attribute:i> <attribute:b>b</attribute:b>a}r</container:p>',
+				'<container:p>Fo<attribute:i>o</attribute:i> ar</container:p>'
+			);
+		} );
+
+		it( 'should remove part of node when range ends inside this node', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>f{oo<attribute:b>ba}r</attribute:b></container:p>',
+				'<container:p>foo<attribute:b>r</attribute:b></container:p>'
+			);
+		} );
+
+		it( 'should remove part of node when range starts inside this node', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>foo<attribute:b>b{ar</attribute:b>bi}z</container:p>',
+				'<container:p>foo<attribute:b>b</attribute:b>biz</container:p>'
+			);
+		} );
+
+		it( 'should remove part of node when range starts and ends inside this node', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>foo<attribute:b>b{a}r</attribute:b>biz</container:p>',
+				'<container:p>foo<attribute:b>br</attribute:b>biz</container:p>'
+			);
+		} );
+
+		it( 'should merge after removing', () => {
+			const elementToRemove = new AttributeElement( 'b' );
+
+			test(
+				elementToRemove,
+				'<container:p>' +
+					'<attribute:a>fo{o</attribute:a><attribute:b>a</attribute:b><attribute:a>b}iz</attribute:a>' +
+				'</container:p>',
+				'<container:p><attribute:a>foobiz</attribute:a></container:p>'
+			);
+		} );
+
+		it( 'should remove EmptyElement', () => {
+			const elementToRemove = new EmptyElement( 'img' );
+
+			test(
+				elementToRemove,
+				'<container:p>f{oo<empty:img></empty:img>ba}r</container:p>',
+				'<container:p>foobar</container:p>'
+			);
+		} );
+
+		it( 'should remove UIElement', () => {
```

Comment:
I think it should work for Containers as well, but  unfortunately because of https://github.com/ckeditor/ckeditor5-engine/issues/802 it doesn't. I'm on it.